orbstation/code/objects/machinery/SAD.dm

```diff
@@ -0,0 +1,267 @@
+	)
+
+/obj/machinery/self_actualization_device/update_appearance(updates)
+	. = ..()
+	if(occupant)
+		icon_state = processing ? "sad_on" : "sad_off"
+	else
+		icon_state = state_open ? "sad_open" : "sad_closed"
+
+
+
+/obj/machinery/self_actualization_device/Initialize(mapload)
+	. = ..()
+	update_appearance()
+
+/obj/machinery/self_actualization_device/close_machine(atom/movable/target, density_to_set = TRUE)
+	..()
+	playsound(src, 'sound/machines/click.ogg', 50)
+	if(!occupant)
+		return FALSE
+	if(!ishuman(occupant))
+		occupant.forceMove(drop_location())
+		set_occupant(null)
+		return FALSE
+	to_chat(occupant, span_notice("You enter [src]."))
+	update_appearance()
+
+/obj/machinery/self_actualization_device/examine(mob/user)
+	. = ..()
+	. += span_notice("ALT-Click to turn ON when closed.")
+
+/obj/machinery/self_actualization_device/open_machine(drop = TRUE, density_to_set = FALSE)
+	playsound(src, 'sound/machines/click.ogg', 50)
+	..()
+
+/obj/machinery/self_actualization_device/AltClick(mob/user)
+	. = ..()
+	if(!powered() || !occupant || state_open)
+		return FALSE
+
+	to_chat(user, "You power on [src].")
+	addtimer(CALLBACK(src, PROC_REF(eject_new_you)), processing_time, TIMER_OVERRIDE|TIMER_UNIQUE)
+	processing = TRUE
+	update_appearance()
+
+/obj/machinery/self_actualization_device/container_resist_act(mob/living/user)
+	if(state_open)
+		open_machine()
+		return FALSE
+
+	to_chat(user, span_notice("The emergency release is not responding! You start pushing against the hull!"))
+	user.changeNext_move(CLICK_CD_BREAKOUT)
+	user.last_special = world.time + CLICK_CD_BREAKOUT
+	user.visible_message(span_notice("You see [user] kicking against the door of [src]!"), \
+		span_notice("You lean on the back of [src] and start pushing the door open... (this will take about [DisplayTimeText(breakout_time)].)"), \
+		span_hear("You hear a metallic creaking from [src]."))
+
+	if(do_after(user, breakout_time, target = src))
+		if(!user || user.stat != CONSCIOUS || user.loc != src || state_open)
+			return
+		user.visible_message(span_warning("[user] successfully broke out of [src]!"), \
+			span_notice("You successfully break out of [src]!"))
+		open_machine()
+
+/obj/machinery/self_actualization_device/interact(mob/user)
+	if(state_open)
+		close_machine()
+		return
+
+	if(!processing)
+		open_machine()
+		return
+
+/obj/machinery/self_actualization_device/process(delta_time)
+	if(!processing)
+		return
+	if(!powered() || !occupant || !iscarbon(occupant))
+		open_machine()
+		return
+
+	next_fact--
+	if(next_fact <= 0)
+		next_fact = rand(initial(next_fact), 2 * initial(next_fact))
+		say(pick(advertisements))
+		playsound(loc, 'sound/machines/chime.ogg', 30, FALSE)
+
+	use_power(500)
+
+/// Ejects the occupant as either their preference character, or as a monke based on emag status.
```

Comment:
This comment does not seem to be true

Reply:
sorry some of this was just ported directly from skyrat code and it seems thats just a goof comment they left on the S.A.D even though it doesn't actually seem to do that.